### PR TITLE
Fixed torchserve e2e test.

### LINKED ISSUE #3094: validation error for InferenceResponse on V2 mnist for v0.11.0
Hi :)

**Environment:**
- Istio Version:  1.17.2
- Knative Version: knative-v1.11.0
- KServe Version: v0.11.0
- Kubernetes version: (use `kubectl version`): v1.25.6

I'm getting error for V2 MNist
**{"error":"ValidationError : 1 validation error for InferenceResponse\nid\n  none is not an allowed value (type=type_error.none.not_allowed)"}**

https://github.com/kserve/kserve/tree/master/docs/samples/v1beta1/torchserve/v2

```
admin@cli:~/kserve$ curl -v -H "Content-Type: application/json" -d @./mnist_v2_tensor.json http://mnist-v2-sa.models.mlde-kserve.abc.lan/v2/models/mnist/infer
*   Trying 10.90.1.50:80...
* Connected to mnist-v2-sa.models.mlde-kserve.abc.lan (10.90.1.50) port 80 (#0)
> POST /v2/models/mnist/infer HTTP/1.1
> Host: mnist-v2-sa.models.mlde-kserve.abc.lan
> User-Agent: curl/7.81.0
> Accept: */*
> Content-Type: application/json
> Content-Length: 6801
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 500 Internal Server Error
< content-length: 141
< content-type: application/json
< date: Mon, 21 Aug 2023 12:06:23 GMT
< server: istio-envoy
< x-envoy-upstream-service-time: 190
<
* Connection #0 to host mnist-v2-sa.models.mlde-kserve.abc.lan left intact
{"error":"ValidationError : 1 validation error for InferenceResponse\nid\n  none is not an allowed value (type=type_error.none.not_allowed)"}
```

```
2023-08-21 11:45:19.628 10 uvicorn.error ERROR [run_asgi():376] Exception in ASGI application
Traceback (most recent call last):
  File "/home/venv/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 373, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/home/venv/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
  File "/home/venv/lib/python3.9/site-packages/fastapi/applications.py", line 270, in __call__
    await super().__call__(scope, receive, send)
  File "/home/venv/lib/python3.9/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/venv/lib/python3.9/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/venv/lib/python3.9/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/venv/lib/python3.9/site-packages/timing_asgi/middleware.py", line 68, in __call__
    await self.app(scope, receive, send_wrapper)
  File "/home/venv/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/home/venv/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/home/venv/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/venv/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/venv/lib/python3.9/site-packages/starlette/routing.py", line 706, in __call__
    await route.handle(scope, receive, send)
  File "/home/venv/lib/python3.9/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/home/venv/lib/python3.9/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/home/venv/lib/python3.9/site-packages/fastapi/routing.py", line 235, in app
    raw_response = await run_endpoint_function(
  File "/home/venv/lib/python3.9/site-packages/fastapi/routing.py", line 161, in run_endpoint_function
    return await dependant.call(**values)
  File "/home/venv/lib/python3.9/site-packages/kserve/protocol/rest/v2_endpoints.py", line 135, in infer
    res = InferenceResponse.parse_obj(response)
  File "pydantic/main.py", line 526, in pydantic.main.BaseModel.parse_obj
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for InferenceResponse
id
  none is not an allowed value (type=type_error.none.not_allowed)
```
  
```
apiVersion: "serving.kserve.io/v1beta1"
kind: "InferenceService"
metadata:
  name: mnist-v2-sa
  namespace: models
spec:
  predictor:
    serviceAccountName: sa
    model:
      modelFormat:
        name: pytorch
      protocolVersion: v2
      storageUri: s3://kmodels/mnist
```

### PR DESCRIPTION
**Pull request by andyi2it**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3094

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?

**Release note**:

```release-note
GRPC v2 support will be added later
```

### CODE REVIEW

File: test/e2e/predictor/test_torchserve.py [hunk 1]
```diff
@@ -51,6 +51,7 @@ def test_torchserve_kserve():
                     "memory": "1Gi"
                 },
             ),
+            env=[V1EnvVar(name="PROTOCOL_VERSION", value="v1")],
```
Comment (yuzisun) [inline test/e2e/predictor/test_torchserve.py]: Is this a breaking change that now you need to add the protocol version environment variable ?

Comment (yuzisun) [reply test/e2e/predictor/test_torchserve.py]: We already have protocol version field on the spec, can we inject the env in the webhook ?

File: pkg/apis/serving/v1beta1/inference_service_defaults.go [hunk 2]
```diff
@@ -310,6 +310,13 @@ func (isvc *InferenceService) SetTorchServeDefaults() {
 	if constants.ProtocolV2 == *isvc.Spec.Predictor.Model.ProtocolVersion {
 		isvc.ObjectMeta.Labels[constants.ServiceEnvelope] = constants.ServiceEnvelopeKServeV2
 	}
+
+	// set torchserve env variable "PROTOCOL_VERSION" based on ProtocolVersion
+	isvc.Spec.Predictor.Model.Env = append(isvc.Spec.Predictor.Model.Env,
+		v1.EnvVar{
+			Name:  constants.ProtocolVersionENV,
+			Value: string(*isvc.Spec.Predictor.Model.ProtocolVersion),
```
Comment (yuzisun) [inline pkg/apis/serving/v1beta1/inference_service_defaults.go:318]: Is it safe the dereference the protocolVersion field here? might need a nil check

Comment (andyi2it) [reply pkg/apis/serving/v1beta1/inference_service_defaults.go:318]: There is an existing nil check at the start of the function, probably wouldn't have shown up in github review as it shows only limited lines.

**Pull request merged**